packages/webapp/src/containers/CropCatalogue/useFilterVarieties.js

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { LOCATION, STATUS, SUPPLIERS } from '../Filter/constants';
+import { cropVarietyFilterSelector } from '../filterSlice';
+
+export default function useFilterVarieties(varieties, cropId, status) {
+  const cropVarietyFilter = useSelector(cropVarietyFilterSelector(cropId));
+  // if (!cropVarietyFilter) return varieties;
+  const filteredBySupplier = useMemo(() => {
+    if (!cropVarietyFilter) return varieties;
+    const supplierFilter = cropVarietyFilter[SUPPLIERS];
+    const included = new Set();
+    for (const supplier in supplierFilter) {
+      if (supplierFilter[supplier].active) included.add(supplier);
+    }
+    if (included.size === 0) return varieties;
+    return varieties.filter((variety) => included.has(variety.supplier));
+  }, [cropVarietyFilter?.[SUPPLIERS], varieties]);
+
+  const filteredByLocation = useMemo(() => {
+    if (!cropVarietyFilter) return filteredBySupplier;
+    const locationFilter = cropVarietyFilter[LOCATION];
+    const included = new Set();
+    for (const location_id in locationFilter) {
+      if (locationFilter[location_id].active) included.add(location_id);
+    }
+    if (included.size === 0) return filteredBySupplier;
+    return varieties.filter((variety) => included.has(variety.location_id));
+  }, [cropVarietyFilter?.[LOCATION], filteredBySupplier]);
+
+  const filteredByStatus = useMemo(() => {
+    if (!cropVarietyFilter) return filteredByLocation;
+    const statusFilter = cropVarietyFilter[STATUS];
+    const activeFilterSuppliers = new Set(
+      Object.keys(statusFilter).filter((status) => statusFilter[status].active),
```

Comment:
Why didn't you do this in the other 2 filters?

Reply:
oh oops let me do that